query/security/common/src/main/java/org/codice/ddf/admin/security/common/fields/wcpm/Realm.java

```diff
@@ -42,25 +45,26 @@ public Realm(ServiceReader serviceReader) {
         this.serviceReader = serviceReader;
     }
 
+    public Realm(ServiceReader serviceReader, EnumValue<String> value) {
+        super(DEFAULT_FIELD_NAME,
+                FIELD_TYPE_NAME,
+                DESCRIPTION);
+        this.serviceReader = serviceReader;
+        setValue(value.value());
+    }
+
     @Override
     public List<EnumValue<String>> getEnumValues() {
-        Set<JaasRealm> realms = serviceReader.getServices(JaasRealm.class, null);
-        return realms.stream().map(realm -> new EnumValue<String>() {
-            @Override
-            public String enumTitle() {
-                return realm.getName();
-            }
-
-            @Override
-            public String description() {
-                return null;
-            }
-
-            @Override
-            public String value() {
-                return realm.getName();
-            }
-        }).collect(Collectors.toList());
+        Set<EnumValuePoller> realms = serviceReader.getServices(EnumValuePoller.class, REALM_POLLER_FILTER);
+
+        if(realms.size() == 0) {
+            return new ArrayList<>();
+        }
+
+        return realms.stream()
```

Comment:
Remove call to `stream()`.

Or simplify this method (and the one in Auth) further if you can make the `EnumValuePoller` of fixed type String:
```java
        Set<EnumValuePoller> realms = serviceReader.getServices(EnumValuePoller.class, REALM_POLLER_FILTER);

        return realms.stream()
                .findFirst()
                .map(EnumValuePoller::getEnumValues)
                .orElse(new ArrayList<>());
```